src/components/Registration/StudentForm.vue

```diff
@@ -0,0 +1,305 @@
+        {{msg}}
+      </div>
+      <div v-else>
+        <table class="step-2-table">
+          <tr>
+            <td class="table-entry" colspan="2"><b>Step 2 of 2: Tell us about yourself! </b></td>
+          </tr>
+          <tr class="question-row">
+            <td class="table-entry" colspan="2">What's your name?</td>
+          </tr>
+          <tr>
+            <td style="padding-right: 15px;"><input class="form-control" required autofocus v-model="profile.firstName"></td>
+            <td style="padding-left: 15px;"><input class="form-control" required autofocus v-model="profile.lastName"></td>
+          </tr>
+          <tr>
+            <td class="table-entry"><div class="description">First Name</div></td>
+            <td class="table-entry"><div class="description">Last Name</div></td>
+          </tr>
+          <tr class="question-row">
+            <td class="table-entry" colspan="2">What high school did you go to?</td>
+          </tr>
+          <tr>
+            <td colspan="2"><input class="form-control" required autofocus v-model="profile.highSchool"></td>
+          </tr>
+          <tr class="question-row">
+            <td class="table-entry" colspan="2">How did you hear about us?</td>
+          </tr>
+          <tr>
+            <td colspan="2">
+              <select class="form-control topic" v-model="profile.heardFrom">
+                <option value="Flyer">Flyer</option>
+                <option value="Email">Email</option>
+                <option value="Internet search">Internet search</option>
+                <option value="Friend">Friend</option>
+                <option value="Family member">Family member</option>
+                <option value="Teacher">Teacher</option>
+                <option value="School">School</option>
+                <option value="Other">Other</option>
+              </select>
+            </td>
+          </tr>
+          <tr>
+            <div class="agreement-box">
+              <input type="checkbox" id="userAgreement" v-model="credentials.terms" required>
+              <label id='agreement' for="userAgreement"></label>
+              <div class="agreement-label">I have read and accept the <a href="#/legal" target="_blank">user agreement</a>.</div>
+            </div>
+          </tr>
+        </table>
+        <button class="btn btn-lg btn-primary btn-block" type="submit" @click.prevent="submit()">SIGN UP</button>
+      </div>
+    </div>
+  </form>
+</template>
+
+<script>
+  import AuthService from 'src/services/AuthService'
+  import UserService from "../../services/UserService";
+
+  export default {
+    data() {
+      return {
+        msg: '',
+        credentials: {
+          email: '',
+          password: '',
+          terms: false
+        },
+        profile: {
+          firstName: '',
+          lastName: '',
+          highSchool: '',
+          heardFrom: ''
+        },
+        step: 1
```

Comment:
Instead of numerical steps with an if/else, use a string enum to match on for each step. This makes identifying and debugging errors much easier. With the current system there is a huge amount that of edge cases which could happen in the event of an error